python/ray/serve/_private/proxy.py

```diff
@@ -1211,9 +1212,17 @@ def __init__(
         self.wrapped_http_proxy = self.http_proxy
 
         for middleware in http_middlewares:
-            self.wrapped_http_proxy = middleware.cls(
-                self.wrapped_http_proxy, **middleware.options
-            )
+            if starlette.__version__ <= "0.34.0":
```

Comment:
This should use `packaging.version` for comparison

Reply:
done.